File: 268. Missing Number/missingNumber.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function(nums) {
    var sort_nums = []
    for (let i=0; i<=nums.length; i++) {
        sort_nums.push(i)
    }
    let set_sort_nums = new Set(sort_nums)
    let set_nums = new Set(nums)
    let diff= new Array([...set_sort_nums].filter(x=>!set_nums.has(x)))
    return diff
};

// Runtime 75 ms / Memory 51.8 MB
// debug: https://jsfiddle.net/

File: 268. Missing Number/missingNumber.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        // pair XOR = 0
        int r = 0;
        for(int num:nums){ r = r^num; }
        for(int i=0; i<=nums.size(); i++){ r = r^i; }
        return r;
    }
};